### PR TITLE
backend/feat/booking_with_vehicle_and_driver_phone_number

### PR DESCRIPTION
**Pull request by AshishSaini25**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

booking_with_vehicle_and_driver_phone_number

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs [hunk 1]
```diff
@@ -410,3 +416,8 @@ clearOnRideStuckDrivers :: ShortId DM.Merchant -> ApiTokenInfo -> FlowHandler Co
 clearOnRideStuckDrivers merchantShortId apiTokenInfo = withFlowHandlerAPI $ do
   checkedMerchantId <- merchantAccessCheck merchantShortId apiTokenInfo.merchant.shortId
   Client.callDriverOfferBPP checkedMerchantId (.drivers.clearOnRideStuckDrivers)
+
+bookingWithVehicleNumberAndPhone :: ShortId DM.Merchant -> ApiTokenInfo -> Common.BookingWithVehicleAndPhoneReq -> FlowHandler APISuccess
```
Comment (hkmangla) [inline Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs:420]: It should be part of Ride APIs. Also, add buildTransaction in this.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs [hunk 2]
```diff
@@ -629,7 +629,7 @@ getNearestDrivers mbVariant LatLong {..} radiusMeters merchantId onlyNotOnRide m
               sedanResult = getResult SEDAN $ variant == SEDAN || (variant == SUV && canDowngradeToSedan)
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs:629]: Needs to handle the same in these functions `buildFullDriverListOnRide`, `buildFullDriverList` and `getNearestDriversCurrentlyOnRide` too

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs [hunk 3]
```diff
@@ -629,7 +629,7 @@ getNearestDrivers mbVariant LatLong {..} radiusMeters merchantId onlyNotOnRide m
...
               hatchbackResult = getResult HATCHBACK $ variant == HATCHBACK || ((variant == SUV || variant == SEDAN) && canDowngradeToHatchback)
               taxiPlusResult = getResult TAXI_PLUS $ variant == TAXI_PLUS
-              taxiResult = getResult TAXI $ variant == TAXI || (variant == TAXI_PLUS && canDowngradeToTaxi)
+              taxiResult = getResult TAXI $ variant == TAXI || (variant == TAXI_PLUS)
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs:632]: This logic is incorrect. It should be 
```
taxiResult = getResult TAXI $ variant == TAXI || (variant == TAXI_PLUS || variant == SUV || variant == SEDAN || variant == HATCHBACK && canDowngradeToTaxi)
```

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 4]
```diff
@@ -895,3 +898,24 @@ updateByPhoneNumber merchantShortId phoneNumber req = do
   pure Success
 
 ---------------------------------------------------------------------
+
+bookingWithVehicleNumberAndPhone :: ShortId DM.Merchant -> Common.BookingWithVehicleAndPhoneReq -> Flow APISuccess
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:902]: This logic doesn't fit this
```
get driverId using phone number
check if driver is register to same vehicle  (by checking inside vehicle table)
     then return
if not,
     then check if vehicle already linked to driver in driver_rc_association , if yes
            then activateRC
     else, addVehicle for that driver
```

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs (hunk 1 above)
Comment (AshishSaini25) [reply Backend/app/dashboard/provider-dashboard/src/API/ProviderPlatform/DynamicOfferDriver/Driver.hs:420]: ok

**Review by hkmangla — APPROVED** [review]

**Pull request merged**